src/mod/features/commands/case_call.cpp

```diff
@@ -0,0 +1,45 @@
+#include "externals/Dpr/EvScript/EvDataManager.h"
+#include "externals/FlagWork_Enums.h"
+#include "externals/FlagWork.h"
+
+#include "features/commands/utils/utils.h"
+#include "logger/logger.h"
+
+
+bool CaseCall(Dpr::EvScript::EvDataManager::Object* manager)
+{
+    Logger::log("_CASE_CALL\n");
+    system_load_typeinfo(0x45dc);
+    EvData::Aregment::Array* args = manager->fields._evArg;
+    int32_t switchWorkIndex = FlagWork::GetWork(manager->fields._switch_work_index);
+    int32_t workNoValue;
+
+    if (args->max_length >= 3)
+    {
+        auto argType = static_cast<EvData::ArgType>(args->m_Items[1].fields.argType);
+
+        if (argType == EvData::ArgType::Work) {
+            workNoValue = args->m_Items[1].fields.data;
+        }
+        else if (argType == EvData::ArgType::Float) {
+            workNoValue = static_cast<float>(args->m_Items[1].fields.data);
+        }
+        else {
+            workNoValue = 0;
+        }
+
+        EvData::Aregment::Object arg3 = args->m_Items[2];
+        System::String::Object* label = System::String::Create("");
+
+        if (static_cast<EvData::ArgType>(arg3.fields.argType) == EvData::ArgType::String) {
```

Comment:
Might be worth putting this part into a util as well, like how GetWorkOrIntValue is done